packages/fast-usdc-contract/src/exos/settler.ts

```diff
@@ -270,29 +270,38 @@ export const prepareSettler = (
 
           const { nfa, amount, EUD } = decoded;
           const { self } = this.facets;
-          const found = statusManager.dequeueStatus(nfa, amount);
-          log('dequeued', found, 'for', nfa, amount);
-          const fullValue = AmountMath.make(USDC, amount);
+          const dequeued = statusManager.matchAndDequeueSettlement(nfa, amount);
 
-          switch (found?.status) {
-            case PendingTxStatus.Advanced:
-              return self.disburse(found.txHash, fullValue, EUD);
+          if (dequeued.length === 0) {
+            log('⚠️ tap: minted before observed', nfa, amount);
+            // XXX consider capturing in vstorage
+            // we would need a new key, as this does not have a txHash
+            self.addMintedEarly(nfa, AmountMath.make(USDC, amount));
+            return;
+          }
 
-            case PendingTxStatus.Advancing:
-              log('⚠️ tap: minted while advancing', nfa, amount);
-              self.addMintedEarly(nfa, fullValue);
-              return;
+          log('dequeued', dequeued, 'for', nfa, amount);
+          for (const found of dequeued) {
+            const fullValue = AmountMath.make(USDC, found.tx.amount);
+            switch (found.status) {
+              case PendingTxStatus.Advanced:
+                void self.disburse(found.txHash, fullValue, EUD);
+                break;
 
-            case PendingTxStatus.AdvanceSkipped:
-            case PendingTxStatus.AdvanceFailed:
-              return self.forward(found.txHash, fullValue, EUD);
+              case PendingTxStatus.Advancing:
+                log('⚠️ tap: minted while advancing', nfa, found.tx.amount);
+                // XXX consider tracking these separately from unknown mints?
+                self.addMintedEarly(nfa, fullValue);
```

Comment:
Is it acceptable to handle other found transactions in this case? If not, it seems like cleanup would need to put some things back.